lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3409,6 +3449,41 @@ ParseResult FIRStmtParser::parseInstance() {
   return moduleContext.addSymbolEntry(id, entryId, startTok.getLoc());
 }
 
+/// object ::= 'object' id 'of' id info?
+ParseResult FIRStmtParser::parseObject() {
+  if (FIRVersion::compare(version, {3, 1, 0}) < 0)
+    return emitError() << "unexpected token: objects are a FIRRTL 3.1.0+ "
+                          "feature, but the specified FIRRTL version was "
+                       << version;
+
+  auto startTok = consumeToken(FIRToken::kw_object);
+
+  // If this was actually the start of a connect or something else handle
+  // that.
+  if (auto isExpr = parseExpWithLeadingKeyword(startTok))
+    return *isExpr;
```

Comment:
Is this necessary? It seems like you should just consume the token as you know it's an object. This should be safe given that this is gated by being in a FIRRTL version which isn't context sensitive around leading expressions/keywords.

Reply:
You raise a good point. I think I need to sink the version check, so it occurs after we try to parse the object decl as an expression, to preserve backwards compatibility.